src/core/bpf-lsm.c

```diff
@@ -48,7 +48,7 @@ static bool bpf_can_link_lsm_program(struct bpf_program *prog) {
         if (!link)
                 return -ENOMEM;
 
-        return 1;
+        return sym_libbpf_get_error(link) == 0;
```

Comment:
maybe add a comment explaining why this is done, given it's not very obvious?

Reply:
This still doesn't look right because `libbpf_get_error` already handles NULL:
```C
long libbpf_get_error(const void *ptr)
{
	if (!IS_ERR_OR_NULL(ptr))
		return 0;

	if (IS_ERR(ptr))
		errno = -PTR_ERR(ptr);

	/* If ptr == NULL, then errno should be already set by the failing
	 * API, because libbpf never returns NULL on success and it now always
	 * sets errno on error. So no extra errno handling for ptr == NULL
	 * case.
	 */
	return -errno;
}
```
Thus, we probably want something like this:
```C
link = sym_bpf_program__attach_lsm(prog);
return sym_libbpf_get_error(link) == 0;
```

Reply:
It would be nice to adjust `bpf_link_free` too:
```diff
- if (!link)
+ if (IS_ERR_OR_NULL(link))
```

Reply:
Oh, IS_ERR_OR_NULL is not public. I'll adjust the comment a bit though.

Reply:
And ` return -ENOMEM;` never worked as intended because function return type is boolean...